common/src/main/java/com/wynntils/features/map/MainMapFeature.java

```diff
@@ -162,32 +156,25 @@ public void onScreenOpened(ScreenOpenedEvent.Post event) {
             return;
         }
 
-        LootChestType chestType = Models.LootChest.getChestType(event.getScreen());
+        LootChestTier chestType = Models.LootChest.getChestType(event.getScreen());
         if (chestType == null) return;
 
         if (chestType.ordinal() < minTierForAutoWaypoint.get().ordinal()) {
             lastChestPos = null;
             return;
         }
 
-        PoiLocation location = new PoiLocation(lastChestPos.getX(), lastChestPos.getY(), lastChestPos.getZ());
-        CustomPoi newPoi = new CustomPoi(
-                location,
-                chestType.getWaypointName(),
-                CommonColors.WHITE,
-                chestType.getWaypointTexture(),
-                CustomPoi.Visibility.DEFAULT);
+        Location location = new Location(lastChestPos.getX(), lastChestPos.getY(), lastChestPos.getZ());
 
-        if (customPois.get().stream().noneMatch(customPoi -> customPoi.equals(newPoi))) {
-            customPois.get().add(newPoi);
+        if (foundChestLocations.get().stream()
```

Comment:
It's kind of weird that we have this in MainMapFeature. It should be in a Model.

Reply:
Should we move both the found chests and custom markers to a (new?) model?

Reply:
Yes and no. :-) They should not live inside a Feature.

However, actual loot chests align with the concept of a Model -- it is something that exists in the Wynncraft world. The user-added waypoints however, are a complete add-on provided by Wynntils, so it is definitely a Service, like favorite items etc.

Reply:
Okay. When migrating from the old customPois config, I'll make sure to migrate to a storage instead.

Reply:
Yep!

Did we decide how to handle the old chests? The problem is that we cannot be 100% sure that the user did not manually enter a custom poi with the same name and icon as an automatically collected chest location. (Perhaps they marked the entrance of a cave that way.)

I think we said something about migrating them to a "legacy" storage, and that when we find a new chest, we add it to a "proper" storage, and also check if it is in the "legacy" storage, and if so, remove it.

Reply:
Then maybe we need to offer a way to handle remaining legacy storage chests after users have been playing for awhile, by either removing them, or moving them to user-added waypoints.

Reply:
I was thinking of moving all loot chests that are of the format we used to the found chests automatically. I think that might be the simplest, but the most comprehensive solution. We can check for the visibility and color too, to rule out custom added ones.